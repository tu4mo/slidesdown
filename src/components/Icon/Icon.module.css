.button {
  background-color: transparent;
  border: 0;
  cursor: pointer;
  display: block;
  padding: 0;
  transition: transform 0.1s ease-in-out;

  &:disabled {
    opacity: 0.1;
  }

  &:hover {
    transform: scale(1.1);
  }

  &:active {
    transform: scale(0.98);
  }
}

.icon {
  display: block;
}
