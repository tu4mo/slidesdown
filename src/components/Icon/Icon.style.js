import styled from 'styled-components'

export const StyledIcon = styled.img.attrs({})`
  cursor: pointer;
  display: block;
  opacity: ${props => (props.disabled ? 0.1 : 1)};
  transition: transform 0.1s ease-in-out;

  &:hover {
    transform: scale(1.1);
  }

  &:active {
    transform: scale(0.98);
  }
`
