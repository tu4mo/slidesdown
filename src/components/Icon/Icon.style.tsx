import styled from 'styled-components'

interface Props {
  disabled: boolean
}

export const StyledIcon = styled.img.attrs({})<Props>`
  cursor: pointer;
  display: block;
  opacity: ${props => (props.disabled ? 0.1 : 1)};
  transition: transform 0.1s ease-in-out;

  &:hover {
    transform: scale(1.1);
  }

  &:active {
    transform: scale(0.98);
  }
`
