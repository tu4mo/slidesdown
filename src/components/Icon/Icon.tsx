import { ComponentProps } from 'react'

import Tooltip from '../Tooltip'

import { StyledButton, StyledIcon } from './Icon.style'

import crossSvg from './svg/cross.svg'
import editSvg from './svg/edit.svg'
import leftSvg from './svg/left.svg'
import maximizeSvg from './svg/maximize.svg'
import minimizeSvg from './svg/minimize.svg'
import presentationSvg from './svg/presentation.svg'
import rightSvg from './svg/right.svg'
import shareSvg from './svg/share.svg'

const ICONS = {
  cross: crossSvg,
  edit: editSvg,
  left: leftSvg,
  maximize: maximizeSvg,
  minimize: minimizeSvg,
  presentation: presentationSvg,
  right: rightSvg,
  share: shareSvg,
}

interface Props {
  alt?: string
  disabled?: boolean
  onClick?(): void
  tooltip?: ComponentProps<typeof Tooltip>['content']
  type: keyof typeof ICONS
}

const Icon = ({ alt, disabled = false, onClick, tooltip, type }: Props) => {
  const icon = (
    <StyledButton
      aria-label={alt}
      disabled={disabled}
      onClick={!disabled ? onClick : undefined}
    >
      <StyledIcon src={ICONS[type]} />
    </StyledButton>
  )

  return tooltip ? <Tooltip content={tooltip}>{icon}</Tooltip> : icon
}

export default Icon
