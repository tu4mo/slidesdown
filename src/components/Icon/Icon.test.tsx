import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Icon from '../Icon'

jest.mock('react-tippy', () => ({
  Tooltip: 'Tooltip'
}))

it('renders correctly', () => {
  const tree = renderer.create(<Icon type="presentation" />).toJSON()

  expect(tree).toMatchSnapshot()
})

it('renders correctly with tooltip', () => {
  const tree = renderer.create(<Icon tooltip="Test" type="share" />).toJSON()

  expect(tree).toMatchSnapshot()
})
