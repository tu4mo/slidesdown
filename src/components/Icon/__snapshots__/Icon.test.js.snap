// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  cursor: pointer;
  display: block;
  opacity: 1;
  -webkit-transition: -webkit-transform 0.1s ease-in-out;
  -webkit-transition: transform 0.1s ease-in-out;
  transition: transform 0.1s ease-in-out;
}

.c0:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<img
  className="c0"
  src="test-file-stub"
/>
`;

exports[`renders correctly with tooltip 1`] = `
.c0 {
  cursor: pointer;
  display: block;
  opacity: 1;
  -webkit-transition: -webkit-transform 0.1s ease-in-out;
  -webkit-transition: transform 0.1s ease-in-out;
  transition: transform 0.1s ease-in-out;
}

.c0:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<Tooltip
  arrow={true}
  html="Test"
>
  <img
    className="c0"
    src="test-file-stub"
  />
</Tooltip>
`;
